### PR TITLE
Feature/variant type support

### PR DESCRIPTION
**Pull request by ShimonSte**
### Summary
Implements full support for Spark 4.0's `VariantType` mapped to ClickHouse's `JSON` type, enabling efficient storage and querying of JSON data. Addresses #404 

### Changes
- **Schema Mapping**: Map ClickHouse `JSON`  <-> Spark `VariantType`
- **Binary Reader**: Handle Java collections from ClickHouse binary protocol
- **JSON Reader**: Parse JSON data into Spark's internal `VariantVal` representation
- **Comprehensive Tests**: 14 test cases covering read/write operations, nested structures, arrays, nulls, and edge cases
- **ClickHouse 25.3**: Upgrade to support native JSON type

### Compatibility
- **Spark Version**: 4.0+
- **ClickHouse Version**: 25.3+
- **Formats**: 
   read -> JSON | Binary
   write -> only JSON (Arrow not supported yet)

### CODE REVIEW

**Comment by mzitnik** [comment]
@ShimonSte, it's not clear 
Formats: JSON | Binary
read -> json/binary 
write -> only JSON

File: spark-4.0/clickhouse-spark/src/main/scala/org/apache/spark/sql/clickhouse/SchemaUtils.scala [hunk 1]
```diff
@@ -29,7 +29,8 @@ object SchemaUtils extends SQLConfHelper {
     val catalystType = chColumn.getDataType match {
       case Nothing => NullType
       case Bool => BooleanType
-      case String | JSON | UUID | Enum8 | Enum16 | IPv4 | IPv6 => StringType
+      case JSON => VariantType
```
Comment (pan3793) [inline spark-4.0/clickhouse-spark/src/main/scala/org/apache/spark/sql/clickhouse/SchemaUtils.scala]: maybe we should have a config to allow users to switch back to the old behavior

```
spark.clickhouse.read.jsonAs=[string|variant]
```

File: spark-4.0/clickhouse-spark/src/main/scala/com/clickhouse/spark/read/format/ClickHouseBinaryReader.scala [hunk 2]
```diff
@@ -32,15 +32,31 @@ import com.clickhouse.data.value.{
...
 import java.io.InputStream
 import java.time.{LocalDate, ZoneOffset, ZonedDateTime}
 import java.util
+
+object ClickHouseBinaryReader {
+  private val jsonMapper = new ObjectMapper()
+  private val jsonFactory = new com.fasterxml.jackson.core.JsonFactory()
+
+  private def buildVariantFromValue(value: Any): org.apache.spark.unsafe.types.VariantVal = {
+    val jsonString = jsonMapper.writeValueAsString(value)
```
Comment (mzitnik) [inline spark-4.0/clickhouse-spark/src/main/scala/com/clickhouse/spark/read/format/ClickHouseBinaryReader.scala:57]: should we use try/catch in case we can not parse it

File: spark-4.0/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala [hunk 3]
```diff
@@ -1585,4 +1585,31 @@ trait ClickHouseReaderTestBase extends SparkClickHouseSingleTest {
     }
   }
 
+  test("decode VariantType - read JSON as string with config") {
```
Comment (mzitnik) [inline spark-4.0/clickhouse-spark-it/src/test/scala/org/apache/spark/sql/clickhouse/single/ClickHouseReaderTestBase.scala:1618]: Maybe we should also have negative test

File: spark-4.0/clickhouse-spark/src/main/scala/com/clickhouse/spark/read/format/ClickHouseBinaryReader.scala (hunk 2 above)
Comment (ShimonSte) [reply spark-4.0/clickhouse-spark/src/main/scala/com/clickhouse/spark/read/format/ClickHouseBinaryReader.scala:57]: We are writing the Binary Variant value from CH as a string and then loading it as a spark Variant.
If it fails here it should through an error (In our case it would fail the task running not the whole spark application)

**Review by mzitnik — APPROVED** [review]
